packages/opentelemetry-plugin-https/test/functionals/https-disable.test.ts

```diff
@@ -42,7 +43,7 @@ describe('HttpsPlugin', () => {
       nock.cleanAll();
       nock.enableNetConnect();
 
-      plugin.enable((https as unknown) as Http, tracer, tracer.logger);
+      plugin.enable((https as unknown) as typeof http, tracer, tracer.logger);
```

Comment:
Do we need this type casting? Can we not do `plugin.enable(https, tracer, tracer.logger)` directly without casts?

Reply:
as of now we need this as the `HttpsPlugin` extends the `HttpPlugin` which is a `BasePlugin<Http>` and therefore requires `typeof http` as input.

Reply:
Why changing for `typeof http` instead of Http. Can you let `Http` for consistence ?

Reply:
@OlivierAlbertini this was actually a bug before. @Flarna is fixing it in this PR. It doesn't make sense to pass the Http plugin into itself. The first parameter is actually the module being instrumented, http in this case.

Reply:
@dyladan perhaps I'm missing something but 
https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-plugin-http/src/types.ts#L44
`Http` is not `HttpPlugin`...
and `Http` = `typeof http`.

Reply:
ah you're correct thanks for clearing that up for me

Reply:
changed to use `Http` instead `typeof http`. Was somehow a side effect during copy stuff without looking too much into the content.